### PR TITLE
AN-1170 add drm metrics to exoPlayer and Bitmovin Player

### PR DESCRIPTION
**Pull request by sabe25**
### Work
- Fixes: https://bitmovin.atlassian.net/browse/AN-1170
- Description: add drm metrics to exoPlayer and Bitmovin Player

### Checklist

- [x] Updated CHANGELOG.md

### CODE REVIEW

File: bitmovinanalyticsexample/src/main/java/com/bitmovin/bitmovinanalyticsexample/MainActivity.java [hunk 1]
```diff
@@ -60,15 +63,16 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-        SourceConfiguration source = this.createSource();
+//        SourceConfiguration source = this.createSource();
+        SourceConfiguration source = this.createDRMSource();
         config.setSourceConfiguration(source);
 
-        AdvertisingConfiguration adConfig = initializeAds(config);
-        config.setAdvertisingConfiguration(adConfig);
+//        AdvertisingConfiguration adConfig = initializeAds(config);
+//        config.setAdvertisingConfiguration(adConfig);
```
Comment (Svarozic) [inline bitmovinanalyticsexample/src/main/java/com/bitmovin/bitmovinanalyticsexample/MainActivity.java:71]: can you put commented block togehter so we now it belongs together ?

File: collector/src/main/java/com/bitmovin/analytics/data/DRMInformation.kt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package com.bitmovin.analytics.data
+
+data class DRMInformation(var loadTime: Long? = null, var type: String? = null)
```
Comment (Svarozic) [inline collector/src/main/java/com/bitmovin/analytics/data/DRMInformation.kt:3]: do we need both props as nullable ? I think instead in Adapter whole `DRMInformation` could be nullable instead.

File: bitmovinanalyticsexample/src/main/java/com/bitmovin/bitmovinanalyticsexample/MainActivity.java (hunk 1 above)
Comment (sabe25) [reply bitmovinanalyticsexample/src/main/java/com/bitmovin/bitmovinanalyticsexample/MainActivity.java]: https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/80/commits/42efad3564b0eef0242516949bcedb2179414237

File: bitmovinanalyticsexample/src/main/java/com/bitmovin/bitmovinanalyticsexample/MainActivity.java [hunk 3]
```diff
@@ -108,9 +112,26 @@ protected BitmovinPlayerCollector initializeAnalytics() {
...
+        // setup DRM handling
+        String drmLicenseUrl = "https://widevine-proxy.appspot.com/proxy";
+        UUID drmSchemeUuid = DRMSystems.WIDEVINE_UUID;
+        sourceItem.addDRMConfiguration(drmSchemeUuid, drmLicenseUrl);
+
+        // Add a new source item
+        sourceConfiguration.addSourceItem(sourceItem);
 
         return sourceConfiguration;
     }
```
[the remark this answers is not in the document]
Comment (sabe25) [reply bitmovinanalyticsexample/src/main/java/com/bitmovin/bitmovinanalyticsexample/MainActivity.java]: https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/80/commits/42efad3564b0eef0242516949bcedb2179414237

File: analyticsexample/src/main/java/com/bitmovin/exoplayeranalyticsexample/MainActivity.java [hunk 4]
```diff
@@ -114,17 +121,52 @@ private void createPlayer() {
...
-            DashChunkSource.Factory source = new DefaultDashChunkSource.Factory(dataSourceFactory);
-            DashMediaSource dashMediaSource = new DashMediaSource.Factory(source, dataSourceFactory).createMediaSource(dashStatic);
+            DashMediaSource dashMediaSource = getMediaSource(dashStatic, dataSourceFactory);
             player.prepare(dashMediaSource);
             player.setPlayWhenReady(true);
 
         }
     }
 
+    private DrmSessionManager<FrameworkMediaCrypto> getDrmSession(String drmLicenseUrl, UUID drmScheme) {
```
[the remark this answers is not in the document]
Comment (sabe25) [reply analyticsexample/src/main/java/com/bitmovin/exoplayeranalyticsexample/MainActivity.java]: https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/80/commits/42efad3564b0eef0242516949bcedb2179414237

File: analyticsexample/src/main/java/com/bitmovin/exoplayeranalyticsexample/MainActivity.java [hunk 5]
```diff
@@ -114,17 +121,52 @@ private void createPlayer() {
...
+
+    private DashMediaSource getMediaSource(Uri dashStatic, DataSource.Factory dataSourceFactory) {
+        DashChunkSource.Factory source = new DefaultDashChunkSource.Factory(dataSourceFactory);
+        DashMediaSource.Factory sourceFactory = new DashMediaSource.Factory(source, dataSourceFactory);
+
+        return sourceFactory.createMediaSource(dashStatic);
+    }
+
+    private HttpMediaDrmCallback createMediaDrmCallback(
+            String licenseUrl, String[] keyRequestPropertiesArray) {
```
[the remark this answers is not in the document]
Comment (sabe25) [reply analyticsexample/src/main/java/com/bitmovin/exoplayeranalyticsexample/MainActivity.java]: https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/80/commits/42efad3564b0eef0242516949bcedb2179414237

File: analyticsexample/src/main/java/com/bitmovin/exoplayeranalyticsexample/MainActivity.java [hunk 6]
```diff
@@ -114,17 +121,52 @@ private void createPlayer() {
...
+                        createMediaDrmCallback(drmLicenseUrl, null);
+                return DefaultDrmSessionManager.newFrameworkInstance(drmScheme, mediaDrmCallback, null);
+            } catch (Exception e ){
+                Log.e("Main Application", e.getMessage());
+            }
+        }
+        return null;
+    }
+
+    private DashMediaSource getMediaSource(Uri dashStatic, DataSource.Factory dataSourceFactory) {
```
[the remark this answers is not in the document]
Comment (sabe25) [reply analyticsexample/src/main/java/com/bitmovin/exoplayeranalyticsexample/MainActivity.java]: https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/80/commits/42efad3564b0eef0242516949bcedb2179414237

File: collector/src/main/java/com/bitmovin/analytics/data/DRMInformation.kt (hunk 2 above)
Comment (sabe25) [reply collector/src/main/java/com/bitmovin/analytics/data/DRMInformation.kt]: https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/80/commits/784c55f4d297a7a6ab20bbfe1f0017164ffb8875
needed to make more changes to accomplish that, but i think it is better this way 👍

File: collector/src/main/java/com/bitmovin/analytics/adapters/PlayerAdapter.java [hunk 7]
```diff
@@ -3,13 +3,13 @@
...
-import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public interface PlayerAdapter {
     EventData createEventData();
 
     void release();
     long getPosition();
-    @NotNull
+    @Nullable
```
Comment (Svarozic) [inline collector/src/main/java/com/bitmovin/analytics/adapters/PlayerAdapter.java:13]: `com/bitmovin/analytics/exoplayer/ExoPlayerAdapter.java` still you wrong `@NotNull` annotation

**Pull request merged**